indico_zodbimport/modules/event_papers.py

```diff
@@ -0,0 +1,224 @@
+            if storage_path is None:
+                self.importer.print_error(cformat('%{red!}File not found on disk; skipping it [{}]')
+                                          .format(convert_to_unicode(old_filename)),
+                                          event_id=self.event.id)
+                continue
+
+            filename = secure_filename(convert_to_unicode(old_filename), 'attachment')
+            content_type = mimetypes.guess_type(old_filename)[0] or 'application/octet-stream'
+            tpl = PaperTemplate(filename=filename, content_type=content_type, size=size,
+                                storage_backend=storage_backend, storage_file_id=storage_path,
+                                name=old_tpl._Template__name, description=old_tpl._Template__description)
+            self.event.paper_templates.append(tpl)
+
+    def _migrate_competences(self):
+        competence_map = {}
+        for avatar, competences in self.pr._userCompetences.viewitems():
+            if avatar.user.id in competence_map:
+                # add to existing list, which SQLAlchemy will commit
+                competence_map[avatar.user.id] += competences
+            elif competences:
+                competence_map[avatar.user.id] = competences
+                competence = PaperCompetence(user=avatar.user, competences=competences)
+                self.event.paper_competences.append(competence)
+
+    def _migrate_paper_roles(self, contribution):
+        for role in contribution.legacy_paper_reviewing_roles:
+            if role.role == PaperReviewingRoleType.referee:
+                contribution.paper_judges.add(role.user)
+            elif role.role == PaperReviewingRoleType.reviewer:
+                contribution.paper_content_reviewers.add(role.user)
+            elif role.role == PaperReviewingRoleType.editor:
+                contribution.paper_layout_reviewers.add(role.user)
+
+    def _migrate_papers(self):
+        contributions = {c.id: c for c in (Contribution.query.with_parent(self.event)
+                                           .options(joinedload('legacy_paper_reviewing_roles')))}
+        for contrib_id, rm in self.pr._contribution_index.iteritems():
+            if contrib_id not in contributions:
+                self.importer.print_warning(cformat('%{yellow!}Contribution {} not found in event').format(contrib_id),
+                                            event_id=self.event.id)
+                continue
+            contribution = contributions[contrib_id]
+            self._migrate_paper_roles(contribution)
+
+
+class EventPapersImporter(LocalFileImporterMixin, Importer):
+    def __init__(self, **kwargs):
+        kwargs = self._set_config_options(**kwargs)
+        super(EventPapersImporter, self).__init__(**kwargs)
+
+    def has_data(self):
+        return PaperFile.query.has_rows() or PaperReview.query.has_rows() or PaperRevision.query.has_rows()
+
+    def migrate(self):
+        self.migrate_event_pr()
```

Comment:
consistency, so far we always have nothing but simple method calls in `migrate()`